File: 2021/01/main.py
import pathlib

def diff(data, lag=1):
    return (b - a for a, b in zip(data,  data[lag:]))

def main(input_path):
    with open(input_path) as fp:
        depths = list(map(int, fp))
    print(sum(d > 0 for d in diff(depths)))
    print(sum(d > 0 for d in diff(depths, lag=3)))

if __name__ == "__main__":
    main(pathlib.Path(__file__).parent.resolve() / "input")

File: 2021/01/main.jl
function main(input_path)
    data = parse.(Int, eachline(input_path))
    println(sum(diff(data) .> 0))
    @views println(sum(data[4:end] - data[1:end-3] .> 0))
end

if abspath(PROGRAM_FILE) == @__FILE__
    main(joinpath(dirname(@__FILE__), "input"))
end
